packages/config/src/validate.ts

```diff
@@ -0,0 +1,48 @@
+import { FlatbreadConfig } from '@flatbread/core';
+
+type ESModule = Record<string, unknown> & { default: Record<string, unknown> };
+
+/**
+ * Validate that the user config has a default export that is an object.
+ *
+ * @param config user config
+ * @returns user config
+ */
+export function isConfigAnESM(config: unknown): config is ESModule {
+  return typeof config === 'object' && config !== null && 'default' in config;
+}
+
+export function validateConfigHasDefaultExport(config: unknown): ESModule {
+  if (isConfigAnESM(config)) {
```

Comment:
do we want to support cjs as well?
```ts
module.exports = defineConfig({ ... })
```

Reply:
I thought about this too and I think we should. I'll have to double check that this works with esbuild but I think it should

Reply:
module.exports = defineConfig won't show up as the default export, it will show up as function instead of an object with a default function

Reply:
Just tested and `module.exports` syntax works in CJS style configs because esbuild transpiles & bundles it to an ESM before it runs :)

So our bases are already covered with this!